apiserver/facades/client/application/deploy.go

```diff
@@ -93,6 +93,10 @@ func DeployApplication(st ApplicationDeployer, model Model, args DeployApplicati
 		logger.Warningf("proceeding with deployment of application %q even though the charm feature requirements could not be met as --force was specified", args.ApplicationName)
 	}
 
+	if corecharm.IsKubernetes(args.Charm) && charm.MetaFormat(args.Charm) == charm.FormatV1 {
+		logger.Warningf("DEPRECATED: %q is a podspec charm, please deploy sidecar charms instead", args.Charm.URL())
```

Comment:
Perhaps: "Deprecation warning: %q is a podspec charm, which will be deprecated in future releases"

Reply:
It's deprecated now, removed in future releases.

`DEPRECATED: %q is a podspec charm, which will be removed in a future release`